public/app/features/serviceaccounts/ServiceAccountsListPage.tsx

```diff
@@ -34,85 +35,153 @@ function mapStateToProps(state: StoreState) {
   };
 }
 
-const mapDispatchToProps = {
-  fetchServiceAccounts,
-  fetchACOptions,
-  updateServiceAccount,
-  removeServiceAccount,
-  setServiceAccountToRemove,
-  changeFilter,
-  changeQuery,
-};
+const connector = connect(mapStateToProps);
 
-const connector = connect(mapStateToProps, mapDispatchToProps);
-
-const ServiceAccountsListPage = ({
-  fetchServiceAccounts,
-  removeServiceAccount,
-  fetchACOptions,
-  updateServiceAccount,
-  setServiceAccountToRemove,
+const ServiceAccountsListPageUnconnected = ({
   navModel,
   serviceAccounts,
   isLoading,
   roleOptions,
   builtInRoles,
-  changeFilter,
-  changeQuery,
   query,
-  filters,
-  serviceAccountToRemove,
+  serviceAccountStateFilter,
 }: Props): JSX.Element => {
+  const dispatch = useDispatch();
   const styles = useStyles2(getStyles);
+  const [isAddModalOpen, setIsAddModalOpen] = useState(false);
+  const [isRemoveModalOpen, setIsRemoveModalOpen] = useState(false);
+  const [isDisableModalOpen, setIsDisableModalOpen] = useState(false);
+  const [newToken, setNewToken] = useState('');
+  const [currentServiceAccount, setCurrentServiceAccount] = useState<ServiceAccountDTO | null>(null);
 
   useEffect(() => {
-    const fetchData = async () => {
-      await fetchServiceAccounts();
-      if (contextSrv.licensedAccessControlEnabled()) {
-        await fetchACOptions();
-      }
-    };
-    fetchData();
-  }, [fetchServiceAccounts, fetchACOptions]);
+    dispatch(fetchServiceAccounts({ withLoadingIndicator: true }));
+    if (contextSrv.licensedAccessControlEnabled()) {
+      dispatch(fetchACOptions());
+    }
+  }, [dispatch]);
```

Comment:
```suggestion
  }, [fetchServiceAccounts, fetchACOptions, contextSrv]);
```
Seems like some dependencies are missing here, also not sure about redux but at least in React `dispatch` has a stable identity so it doesn't need to be tracked in hooks' dependencies.